code/game/atoms.dm

```diff
@@ -260,17 +260,27 @@
 			if(reagents.reagent_list.len)
 				if(user.can_see_reagents()) //Show each individual reagent
 					for(var/datum/reagent/R in reagents.reagent_list)
-						to_chat(user, "[R.volume] units of [R.name]")
+						if(R.volume < 1000)
+							to_chat(user, "[R.volume] units of [R.name]")
+						else
+							to_chat(user, "[round(R.volume / 1000, 0.01)] kilounits of [R.name]")
 				else //Otherwise, just show the total volume
 					var/total_volume = 0
 					for(var/datum/reagent/R in reagents.reagent_list)
 						total_volume += R.volume
-					to_chat(user, "[total_volume] units of various reagents")
+					if(total_volume < 1000)
+						to_chat(user, "[total_volume] units of various reagents")
+					else
+						to_chat(user, "[round(total_volume / 1000, 0.01)] kilounits of various reagents")
 			else
 				to_chat(user, "Nothing.")
 		else if(container_type & AMOUNT_VISIBLE)
 			if(reagents.total_volume)
-				to_chat(user, "<span class='notice'>It has [reagents.total_volume] unit\s left.</span>")
+				if(reagents.total_volume < 1000)
+					to_chat(user, "<span class='notice'>It has [reagents.total_volume] unit\s left.</span>")
+				else
+					to_chat(user, "<span class='notice'>It has [round(reagents.total_volume / 1000, 0.01)] kilounits left.</span>")
+

```

Comment:
replace lines 279 - 282 with:

to_chat(user, "[total_volume%1000]?"[round(total_volume / 1000, 0.01)] kilounits":"[total_volume] units" of various reagents")